### PR TITLE
Better checks for seo_tag.image

### PR DESCRIPTION
**Pull request by LorenzBung**
References #389

### CODE REVIEW

**Comment by ashmaroli** [comment]
Have you tested this, manually at least?
`first` as a filter is not invoked with the dot operator (`.`) but via the pipe operator (`|`)

**Comment by LorenzBung** [comment]
Yes I have, from [the documentation](https://shopify.dev/docs/themes/liquid/reference/filters/array-filters#first):

> You can use last with dot notation when you need to use the filter inside a tag.

**Comment by ashmaroli** [comment]
Okay I concede. But do note that the Liquid you're referring to is for Shopify themes, Jekyll uses the version as referenced by [this documentation](https://shopify.github.io/liquid/filters/first/).

**Comment by LorenzBung** [comment]
You're right about that, I'll keep that in mind

**Review by Sena — APPROVED** [review]

File: lib/template.html [hunk 1]
```diff
@@ -29,13 +29,19 @@
...
+    {% if seo_tag.image.width %}
+      <meta property="og:image:width" content="{{ seo_tag.image.width }}" />
+    {% endif %}
+    {% if seo_tag.image.height %}
+      <meta property="og:image:height" content="{{ seo_tag.image.height }}" />
+    {% endif %}
   {% endif %}
-  {% if seo_tag.image.width %}
-    <meta property="og:image:width" content="{{ seo_tag.image.width }}" />
+{% else %}
```
Comment (shushugah) [inline lib/template.html:42]:
```
{% elsif seo_tag.image %}
```

File: lib/template.html [hunk 2]
```diff
@@ -29,13 +29,19 @@
   <meta property="og:site_name" content="{{ seo_tag.site_title }}" />
 {% endif %}
 
-{% if seo_tag.image %}
-  <meta property="og:image" content="{{ seo_tag.image.path }}" />
-  {% if seo_tag.image.height %}
-    <meta property="og:image:height" content="{{ seo_tag.image.height }}" />
+{% if seo_tag.image.first %}
```
Comment (shushugah) [inline lib/template.html:32]: `{% if seo_tag.image.first and seo_tag.image.path %}`

**Review by shushugah — COMMENTED** [review]
suggestion to combine nested if/else conditions

**Comment by ashmaroli** [comment]
@LorenzBung This needs tests.
It is not clear what `seo_tag.image.first` returns nor the purpose of checking `seo_tag.image.first` when it is not subsequently used.

Also, I see a lot of `seo_tag.image` calls now. I think you could stash that in a variable and reference the variable thereafter..
```liquid
{% if seo_tag.image %}
  {% assign abcd = seo_tag.image %}
  <meta property="og:image" content="{{ abcd.path }}" />
  {% if abcd.width %}
    <meta property="og:image:width" content="{{ abcd.width }}" />
  {% endif %}
  ...
{% endif %}
```

**Comment by LorenzBung** [comment]
> It is not clear what `seo_tag.image.first` returns nor the purpose of checking `seo_tag.image.first` when it is not subsequently used.

The value it returns is irrelevant here. I just want to check whether `seo_tag.image` contains a string (=> `seo_tag.image` is the image URL itself) or if it is a dictionary, (=> `seo_tag.image.path` contains the URL).
In the latter case, `seo_tag.image.first` will contain some value which is why I check it.

> Also, I see a lot of seo_tag.image calls now. I think you could stash that in a variable and reference the variable thereafter..

Yeah I could but honestly I found it quite readable the way it currently is - if you want I can add some variable though.

**Comment by ashmaroli** [comment]
> The value it returns is irrelevant here..

I took a peak under the hood and the tests:
https://github.com/jekyll/jekyll-seo-tag/blob/22a160cb32681a4349db7105626ad6aa0b1b2954/lib/jekyll-seo-tag/drop.rb#L106-L111
https://github.com/jekyll/jekyll-seo-tag/blob/22a160cb32681a4349db7105626ad6aa0b1b2954/spec/jekyll_seo_tag/image_drop_spec.rb#L17-L22

So basically, if a `page` has `image: foo.png` in the front matter, then `seo_tag.image` results in a truthy.
Therefore, I think this pull request is unnecessary entirely..

**Comment by LorenzBung** [comment]
> So basically, if a `page` has `image: foo.png` in the front matter, then `seo_tag.image` results in a truthy.
> Therefore, I think this pull request is unnecessary entirely..

The reason for this PR is that `image: foo.png` would not be added as meta tag correctly because the liquid template expects `image.path: foo.png`.

**Comment by ashmaroli** [comment]
> The reason for this PR is that `image: foo.png` would not be added as meta tag correctly

@LorenzBung Can you demonstrate this with a test-site?
Since this plugin is whitelisted on GitHub Pages, you will be able to demo easily as well..
Just create a new site locally with `jekyll new ...` and add `image: foo.png` to `about.md` and `image.path: foo.png` in a new page named `test.md`.
Push the site onto GitHub and enable GitHub Pages builds on the repo.

**Comment by LorenzBung** [comment]
Hm, weird, all my tries to reproduce this bug were not successful. The only thing I observed was `image.path: foo.png` not resulting in any output, while `image: foo.png` and `image:\n    path: foo.png` did.

Sorry for wasting your time.

**Comment by ashmaroli** [comment]
> The only thing I observed was `image.path: foo.png` not resulting in any output, while `image: foo.png` and `image:\n path: foo.png` did.

The reason is `image.path` is not a valid data writer. It is a reference to the following YAML front matter:
```
---
image:
  path: foo.png
---
```

**Pull request closed without merging**